include/parameter.h

```diff
@@ -115,7 +115,7 @@ class ParameterBase : public Object {
     }
 
     friend bool operator!=(const ParameterBase<T>& first, const ParameterBase<T>& second) {
-        return !(first == second);
+        return &first != &second;
```

Comment:
I think it was an unhelpful `clang-tidy` suggestion. However, I only implemented this method at all because the implicit definition was causing a problem that I can no longer reproduce. I removed it and it seems to be working correctly again.